lib/livebook/hubs/dockerfile.ex

```diff
@@ -157,6 +157,42 @@ defmodule Livebook.Hubs.Dockerfile do
     |> Enum.join("\n")
   end
 
+  @doc """
+  Builds Dockerfile definition for Livebook Agent app deployment.
+  """
+  @spec build_agent_dockerfile(config(), Hubs.Provider.t()) :: String.t()
+  def build_agent_dockerfile(config, hub) do
+    base_image = Enum.find(Livebook.Config.docker_images(), &(&1.tag == config.docker_tag))
+
+    image = """
+    FROM ghcr.io/livebook-dev/livebook:#{base_image.tag}
+    """
+
+    image_envs = format_envs(base_image.env)
+
+    hub_config = """
+    # Teams Hub configuration for Livebook Agent deployment
+    ENV LIVEBOOK_AGENT_NAME ""
+    ENV LIVEBOOK_TEAMS_KEY "#{hub.teams_key}"
+    ENV LIVEBOOK_TEAMS_AUTH "online:#{hub.hub_name}:#{hub.org_id}:#{hub.org_key_id}:${LIVEBOOK_AGENT_KEY}"
```

Comment:
For this to work `LIVEBOOK_AGENT_KEY` needs to be set at build time anyway, because the `ENV` is defined at build time.

If we want to stick to a Dockerfile I think having LIVEBOOK_AGENT_NAME, LIVEBOOK_AGENT_KEY, LIVEBOOK_TEAMS_KEY as build args is ok. My main point was that in this case we don't need a Dockerfile at all, because we are not really changing the image, so for the end user it would be simpler if we listed image name(s) and the env vars they need to set. But again, up to @josevalim :)